### ISSUE TITLE
Improve standard error approximation for continuous learners

### ISSUE THREAD

**Issue by RaphaelS1** [mlr-org/mlr3proba#53]
Similarly to #52 the current standard error approximation over all time-points is fine for discrete learners but important properties may not hold for continuous learners. However the best approximation is not obvious and is probably worth discussion before implementation (unless anyone has some clear ideas for how to do this already).

The problem is that aggregation of standard error over time-points is non-trivial as time-points are correlated - therefore a sample mean doesn't work. We approximate the error by instead taking the mean over the sample covariance matrix. `se(L) = sqrt(sum(cov(times))/(NM^2)` where `M` is the number of unique time-points. For discrete learners this approximation works well, however for continuous learners, `M` grows much faster than `N`  (as each new observation introduces a new survival time) and quickly stops working as a good approx.

**Comment by fkiraly** [comment]
Just to update this issue with information from off-line discussion:

The formula ```se(L) = sqrt(sum(cov(times))/(NM^2)``` is a plug-in estimate for the standard error of the expected generalization loss of a *fitted* predictor, that can be derived for the case where in the data frame/matrix called ```times``` above, (i) there are ```N``` statistically independent rows, and (ii) have fixed length (number of columns/variables) of ```M```.

We have the problem that (i) for the weighted integrated losses, neither are the rows independent (association through weighting factors), and (ii) the number of columns grows with test set size (number ot test points).

(ii) can be solved for example through not making any correlation correction and treating the test losses as independent; (i) could be solved by obtaining weighting factors from the training set, but that causes a potential problem of divide-by-zero-weight.

Problem (i), more generally, does not occur for unweighted instances, or instances where weights are independent of the test set.

In summary, this is perhaps a theory question to park for the time being.
